documentation-website/src/components/language-switch.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {
+  lazy, Suspense,
+  useState,
+} from 'react';
+import languages from '../locales/languages';
+import Window from './window.ts';
+import './language-switch.scss';
+import t from './t.ts';
+
+const that: Window = window as unknown as Window;
+
+const LanguageSwitch = () => {
+  const [
+    language,
+    setLanguage,
+  ] = useState<string>(() => that?.localStorage?.getItem('language',) ?? 'en',);
+
+  const changeLanguage = (newLanguage: string,) => {
+    setLanguage(newLanguage,);
+    that?.localStorage?.setItem('language', newLanguage,);
+    // reload page
+    that?.location?.reload();
+  };
+
+  const EL = lazy(async() => {
+    const ariaLabel = await t('language-switch.aria',);
+    return {
+      default: () => <select
+        className='language-switch'
+        aria-label={ ariaLabel }
+        value={ language }
+        onChange={ (event,) => changeLanguage(event.target.value,) }
+      >{ languages.map((lang,) => <option
+          key={ lang }
+          value={ lang }
+        >{ lang }</option>,) }</select>,
```

Comment:
maybe refer to
```tsx
<Lang lnkey={`languages.${ lang }` as languageKey}/>
```
here?

Reply:
<img width="120" alt="Screenshot 2024-04-23 at 16 10 48" src="https://github.com/Idrinth/api-bench/assets/116084393/0d8b211f-3d17-4548-b829-26dc7ce6606e">
is it good like this?

Reply:
perfect!